bokeh/core/properties.py

```diff
@@ -636,6 +636,9 @@ def choose_default():
             return self._type_params[0]._raw_default()
         default = kwargs.get("default", choose_default)
         super(Either, self).__init__(default=default, help=help)
+        self.alternatives = []
+        for tp in self._type_params:
+            self.alternatives.extend(tp.alternatives)
```

Comment:
This change expands `Either` to consider any `.accept` calls that its type params may have made